### PR TITLE
feat: add --remove-orphaned flag to mani sync command

### PR DESCRIPTION
**Pull request by fazzabi**
### What's Changed

This PR implements the `--remove-orphaned` flag for the `mani sync` command, addressing the feature request in issue #108. The implementation provides a safe way to clean up project directories that are no longer defined in the mani configuration.

**Key Features:**
- **Safety-first approach**: Only removes directories containing `.git` folders (git repositories)
- **User confirmation**: Shows a preview of what will be deleted and requires explicit confirmation
- **Configuration option**: Supports `remove_orphaned: true` in mani.yaml for permanent enablement
- **Comprehensive testing**: Unit tests cover edge cases and integration scenarios

**Example Usage:**
```bash
# Use flag to remove orphaned projects with confirmation
mani sync --remove-orphaned

# Or enable permanently in mani.yaml
remove_orphaned: true
```

### Technical Description

**Core Implementation:**
- **Git-repository-only detection**: The implementation uses `os.Stat(filepath.Join(path, ".git"))` to identify git repositories, ensuring non-git directories are never removed
- **Path resolution**: Uses `core.GetAbsolutePath()` for consistent path handling across different project configurations
- **User confirmation**: Implements `bufio.NewReader(os.Stdin)` for interactive confirmation with clear preview of deletion targets
- **Error handling**: Comprehensive error wrapping with `fmt.Errorf()` and graceful failure modes

**Safety Measures:**
1. **Whitelist approach**: Only git repositories are considered for removal
2. **Explicit confirmation**: Defaults to 'No' unless user explicitly types 'y' or 'yes'
3. **Preview display**: Shows relative paths of directories that will be removed
4. **Graceful cancellation**: Operation can be cancelled at any point without side effects

**Architecture:**
- `RemoveOrphanedProjects()` - Public API with confirmation
- `removeOrphanedProjectsWithConfirm()` - Internal function supporting test scenarios
- Flag integration follows existing patterns in `cmd/sync.go`
- Configuration parsing in `core/dao/config.go` with safe defaults

**Testing Strategy:**
- Unit tests verify git-repository-only detection logic
- Edge case coverage: non-git directories, hidden directories, active projects
- Integration test ensures flag works in real scenarios
- Mock-friendly design allows testing without user interaction

> Before opening a Pull Request you should read and agreed to the Contributor Code of Conduct (see `CONTRIBUTING.md`)

### CODE REVIEW

File: cmd/sync.go [hunk 1]
```diff
@@ -134,13 +139,23 @@ func runSync(
...
 		if *config.SyncGitignore {
 			err := exec.UpdateGitignoreIfExists(config)
 			core.CheckIfError(err)
 		}
 
 		err = exec.CloneRepos(config, projects, syncFlags)
 		core.CheckIfError(err)
+
+		// Handle orphaned project removal
+		if *config.RemoveOrphaned || syncFlags.RemoveOrphaned {
```
Comment (alajmo) [inline cmd/sync.go:155]: Redudant syncFlags.RemoveOrphaned check here: 

```
if *config.RemoveOrphaned {
	err = exec.RemoveOrphanedProjects(config, projects)
	core.CheckIfError(err)
}
```

should be enough, since the `*config.RemoveOrphaned` has a value always, either defaults to `false` or `true` if set in the mani.yaml.

And if flag is used, the that will set the config file, as u added it above:

```
if setSyncFlags.SyncGitignore {
	config.SyncGitignore = &syncFlags.SyncGitignore
}
```

File: core/exec/clone.go [hunk 2]
```diff
@@ -397,3 +398,123 @@ func PrintProjectInit(projects []dao.Project) {
 	fmt.Println()
 	print.PrintTable(data.Rows, options, data.Headers, []string{}, os.Stdout)
 }
+
+// RemoveOrphanedProjects removes project directories that are no longer defined in the mani configuration
+func RemoveOrphanedProjects(config *dao.Config, projects []dao.Project) error {
+	return removeOrphanedProjectsWithConfirm(config, projects, true)
+}
+
+// removeOrphanedProjectsWithConfirm is the internal function that can skip confirmation for testing
```
Comment (alajmo) [inline core/exec/clone.go:407]: I don't like to add functions that are just necessary for testing, better to just pass in projects instead of config.projects to RemoveOrphanedProjects.

Comment (fazzabi) [reply core/exec/clone.go]: ✅ **Fixed** in commit 792ad41

Removed the `removeOrphanedProjectsWithConfirm` helper function entirely. The logic is now consolidated into a single `RemoveOrphanedProjects` function that always requires user confirmation:

- ❌ Removed: `removeOrphanedProjectsWithConfirm()` test-only function
- ✅ Simplified: Single function with confirmation built-in
- 🧪 Updated tests to only verify the identification logic, with actual removal tested via integration tests

The function is now cleaner and doesn't have test-specific code paths.

**Comment by fazzabi** [comment]
@alajmo Thank you for the thorough review! I've addressed all three comments in commit 792ad41:

1. **Removed redundant flag check** - Now using only `*config.RemoveOrphaned` as suggested
2. **Cleaned up function signature** - Removed unused `projects` parameter 
3. **Eliminated test-only function** - Consolidated into single function with built-in confirmation

The functionality has been thoroughly tested manually and all existing tests pass. The feature now has a cleaner implementation while maintaining the same safety guarantees and user experience.

Ready for re-review! 🚀

**Comment by fazzabi** [comment]
Hi @alajmo ! Just a gentle reminder to review this PR when you have a moment. Let me know if you have any questions or need more details. Thank you!
